internal/chain/engine.go

```diff
@@ -292,6 +296,7 @@ func (e *engineImpl) Finalize(
 				)
 				state.AddBalance(delegation.DelegatorAddress, totalWithdraw)
 			}
+			wrapper.LastEpochInCommittee = newShardState.Epoch
```

Comment:
This is not the right way. LastEpochInCommittee should only be set if the validator is in the shardState.

Reply:
Hmm, that’s how it was before I merged big slashing PR

Reply:
No, it's not, just see the deleted code on the left side, it was ranging over shardState.validators. Now it's ranging over all validators.

Reply:
You're right, just looked a bit closer on older version code: 

```
 git show 801b4b83e277bfc50600d448fed3774cd4864650~1:internal/chain/engine.go
```